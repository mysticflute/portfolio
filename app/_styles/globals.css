/* The file contains root-level custom properties and element resets only. */

:root {
  /* colors */
  --color-gray-100: #0b0b0b;
  --color-gray-200: #393939;
  --color-gray-300: #5e5e5f;
  --color-gray-400: #878787;
  --color-neutral-100: #fff;
  --color-neutral-200: #eff0f6;
  --color-neutral-300: #eceae8;
  --color-red-100: #e85d75;
  /* --color-red-100: #e574bc; */
  --color-yellow-100: #f6b468;
  --color-blue-100: #53abf8;
  --color-purple-100: #4331f0;
  --color-purple-200: #5358f8;
  --color-primary: var(--color-purple-100);
  --color-accent-100: var(--color-purple-200);
  --color-accent-200: var(--color-blue-100);
  --color-accent-300: var(--color-yellow-100);
  --color-accent-400: var(--color-red-100);
  --color-background-100: var(--color-gray-100);
  --color-background-200: var(--color-neutral-100);
  --color-background-300: var(--color-gray-100);
  --color-background-highlight: var(--color-blue-100);
  --color-text: var(--color-gray-200);
  --color-text-accent: var(--color-gray-100);
  --color-text-alt: var(--color-gray-100);
  --color-text-link: var(--color-gray-100);
  --color-text-highlight: var(--color-blue-100);
  --color-text-inverse: var(--color-neutral-200);

  /* spacing */
  --layout-max-width: 1300px;
  --layout-padding-x: 24px;
  --layout-padding-y: 0;
  --layout-padding: var(--layout-padding-y) var(--layout-padding-x);
  --layout-margin-x: auto;
  --layout-margin-y: 0;
  --layout-margin: var(--layout-margin-y) var(--layout-margin-x);
}

*,
::before,
::after {
  box-sizing: border-box;
}

html {
  scroll-behavior: smooth;
  scroll-padding: 2rem;
}

body {
  color: var(--color-text);
  background: var(--color-primary);
  font-family: var(--font-default), system-ui, sans-serif;
  font-weight: 500;
  font-size: 1.125rem;
  line-height: 1.667em;
}

@media screen and (max-width: 767px) {
  body {
    font-size: 1rem;
  }
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  font-size: 1rem;
}

ol,
ul {
  margin: 0;
  padding: 0;
  list-style: none;
}

p {
  margin: 0;
  padding: 0;
}

audio,
canvas,
iframe,
img,
svg,
video {
  vertical-align: middle;
}

img {
  max-width: 100%;
  height: auto;
}

a {
  color: inherit;
  text-decoration: none;
}

@media screen and (max-width: 991px) {
  :root {
    --layout-max-width: calc(600px + var(--layout-padding-x) * 2);
  }
}

@media screen and (max-width: 767px) {
  :root {
    --layout-max-width: calc(500px + var(--layout-padding-x) * 2);
  }
}

@media screen and (max-width: 479px) {
  :root {
    --layout-padding-x: 20px;
  }
}
