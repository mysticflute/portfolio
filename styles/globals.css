:root {
  --color-dark: #0b0b0b;
  --color-blue: #1c92ff;
  --color-red: #ff4a60;
  --color-yellow: #ffc431;
  --color-text-default: #393939;
  --color-text-dark: var(--color-dark);
  --color-text-header-link: var(--color-dark);
  --color-text-header-link-highlight: var(--color-blue);
  --color-background-default: #faf9f6;
  --color-background-dark: var(--color-dark);
  --color-background-highlight: var(--color-blue);
}

*,
::before,
::after {
  box-sizing: border-box;
}

body {
  color: var(--color-text-default);
  background: var(--color-background-default);
  font-family: var(--font-default), system-ui, sans-serif;
  font-weight: 500;
  font-size: 1.125rem;
  line-height: 1.667em;
}

@media screen and (max-width: 767px) {
  body {
    font-size: 1rem;
  }
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  font-size: 1rem;
}

ol,
ul {
  margin: 0;
  padding: 0;
  list-style: none;
}

p {
  margin: 0;
  padding: 0;
}

audio,
canvas,
iframe,
img,
svg,
video {
  vertical-align: middle;
}

img {
  max-width: 100%;
  height: auto;
}

a {
  color: inherit;
  text-decoration: none;
}

/* @media (prefers-color-scheme: dark) {
  html {
    color-scheme: dark;
  }
} */

/* @media (prefers-color-scheme: dark) {
  :root {
  }
} */
