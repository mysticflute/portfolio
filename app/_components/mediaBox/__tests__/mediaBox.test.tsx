import { describe, expect, it } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import Image from 'next/image';
import testImage from '@/public/images/icons/pulsar/services.svg';
import MediaBox from '../mediaBox';

describe('mediaBox', () => {
  it('matches snapshot for default options', () => {
    const { container } = render(
      <MediaBox
        heading="Test Heading"
        image={<Image src={testImage} alt="Test Image" />}
      >
        Test Content
      </MediaBox>,
    );

    expect(container).toMatchSnapshot();
  });

  it('uses the specified heading level', () => {
    render(
      <MediaBox
        heading="Test Heading"
        headingLevel={2}
        image={<Image src={testImage} alt="Test Image" />}
      >
        Test Content
      </MediaBox>,
    );

    expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
  });

  it('adds the correct class name for vertical orientation', () => {
    render(
      <MediaBox
        heading="Test Heading"
        image={<Image src={testImage} alt="Test Image" />}
        orientation="vertical"
        data-testid="test-box"
      >
        Test Content
      </MediaBox>,
    );

    expect(screen.getByTestId('test-box')).toHaveClass('vertical');
    expect(screen.getByTestId('test-box')).not.toHaveClass('horizontal');
  });

  it('adds the correct class name for horizontal orientation', () => {
    render(
      <MediaBox
        heading="Test Heading"
        image={<Image src={testImage} alt="Test Image" />}
        orientation="horizontal"
        data-testid="test-box"
      >
        Test Content
      </MediaBox>,
    );

    expect(screen.getByTestId('test-box')).toHaveClass('horizontal');
    expect(screen.getByTestId('test-box')).not.toHaveClass('vertical');
  });

  it('adds the custom class name', () => {
    render(
      <MediaBox
        heading="Test Heading"
        image={<Image src={testImage} alt="Test Image" />}
        className="customclass"
        data-testid="test-box"
      >
        Test Content
      </MediaBox>,
    );

    expect(screen.getByTestId('test-box')).toHaveClass('customclass');
  });
});
