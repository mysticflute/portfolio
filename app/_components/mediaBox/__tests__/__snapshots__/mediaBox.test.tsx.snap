// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mediaBox matches snapshot for default options 1`] = `
<div>
  <div
    class="box flat box horizontal"
  >
    <div
      class="image"
    >
      <img
        alt="Test Image"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
    <div
      class="content"
    >
      <h3
        class="textHeadingXXSmall"
      >
        Test Heading
      </h3>
      Test Content
    </div>
  </div>
</div>
`;
